### ISSUE TITLE
ReplyButton's Action will not be called?

### ISSUE THREAD

**Issue by guest6379** [tucnak/telebot#259]
the code below will not work
```golang
replyBtn := tb.ReplyButton{
  Text: "🌕 Button #1",
  Action: func(*Callback) { // may not be called??
    fmt.Println("button clicked")
  },
}

b.Handle(&replyBtn, replyBtn.Action)
```

> It may work if the codes
https://github.com/tucnak/telebot/blob/a67e593e355fd1e6cfcc864b48499bc0e9f1ce5c/options.go#L133

is replaced with 
 ```golang 
Action func(*Message) `json:"-"`
```
so we may not register like this:
```golang
b.Handle(&replyBtn, func(m *tb.Message) {
		// on reply button pressed
	})
```

**Comment by demget** [comment]
I'm going to delete `Action` field from reply and inline button types in `v2.1`. It's not used for anything in the package, instead it confuses people which are just started to use telebot.

So it's better for you to handle buttons in traditional way :)

**Comment by guest6379** [comment]
> I'm going to delete `Action` field from reply and inline button types in `v2.1`. It's not used for anything in the package, instead it confuses people which are just started to use telebot.
> 
> So it's better for you to handle buttons in traditional way :)

I don't think it's a good idea to delete `Action`.

For some reasons, we need anonymous callback to bind to reply and inline button, especiallly to the button which be created dynamically. 

Action field with the anonymous callback can make codes to be more pretty and short, isn't it?


```golang

func doAction(who string) {
 // TODO
}


// ............
menu := make([][]tb.InlineButton, 0)
// ...........
for _, name := range suggestNames {
    who := name
    tmpButton := tb.InlineButton{
        Unique: generateUUID(),
        Text:   name,
        Action: func(callback *tb.Callback) {
            doAction(who)
        },
    }
    menu = append(menu, []tb.InlineButton{tmpButton})
    b.Handle(&tmpButton, tmpButton.Action)
}
```

we love `Action`, make it better rather than drop it  forever :smile:

**Comment by demget** [comment]
Isn't it the same?

```go
func doAction(who string) {
 // TODO
}

menu := make([][]tb.InlineButton, 0)
for _, name := range suggestNames {
    who := name
    btn := tb.InlineButton{
        Unique: generateUUID(),
        Text:   name,
    }
    menu = append(menu, []tb.InlineButton{btn})
    b.Handle(&btn, func(callback *tb.Callback) {
        doAction(who)
    })
}
```

**Comment by guest6379** [comment]
how about compatibility of the code used `Action` before?
how about the feature/improvement which may automatically register `Action` callback to bot handler?

**Comment by demget** [comment]
> how about compatibility of the code used `Action` before?

I agree with you about backwards compatibility. The problem of `Action` is that it actually useless. It seems it was left for some unimplemented feature.

Then ok, I'll fix `ReplyButton`'s `Action` and leave it for `v2.1` at least.

**Comment by guest6379** [comment]
> > how about compatibility of the code used `Action` before?
> 
> I agree with you about backwards compatibility. The problem of `Action` is that it actually useless. It seems it was left for some unimplemented feature.
> 
> Then ok, I'll fix `ReplyButton`'s `Action` and leave it for `v2.1` at least.

It sounds good, thanks. Btw,  which version of Telegram Bot API is related to `v2.1` , since the latest Bot API is 4.7 now! https://core.telegram.org/bots/api#january-23-2020

**Comment by demget** [comment]
@guest6379 Bot API 4.7 will be implemented in `v2.1`, which is coming soon!

**Comment by guest6379** [comment]
> @guest6379 Bot API 4.7 will be implemented in `v2.1`, which is coming soon!

Great, thank you so much for the amazing work
